### ISSUE TITLE
Texture.from ICanvas type Issue in v7.0.0-beta.2

### ISSUE THREAD

**Issue by bigtimebuddy** [pixijs/pixijs#8721]
Was upgrading to the latest beta.2 and got this error building https://github.com/pixijs/gif

```
src/AnimatedGIF.ts:277:37 - error TS2345: Argument of type 'HTMLCanvasElement' is not assignable to parameter of type 'TextureSource | TextureSource[]'.
  Type 'HTMLCanvasElement' is missing the following properties from type 'HTMLVideoElement': disablePictureInPicture, onenterpictureinpicture, onleavepictureinpicture, playsInline, and 54 more.

277         this.texture = Texture.from(canvas, { scaleMode });
```

This now errors:
```ts
const canvas = document.createElement('canvas');
const texture = Texture.from(canvas, { scaleMode });
```

Casting works:
```ts
const canvas = document.createElement('canvas') as ICanvas;
const texture = Texture.from(canvas, { scaleMode });
```

This doesn't seem correct to me. WDYT @SuperSodaSea?

**Comment by SuperSodaSea** [comment]
It seems that this error is caused by `strictNullChecks`. PixiJS itself use `strictNullChecks: false`, so I didn't recognize it out earlier. Will fix in #8726.

(By the way, I tried to set `strictNullChecks` to `true` in PixiJS and then I got TONS of errors... I hope we can gradually reduce these errors, and eventually enable `strictNullChecks` at some point in the future?)

**Comment by bigtimebuddy** [comment]
Yes ,(and probably all libraries) should be working toward "strict as possible" with types. This is a recent hold-over from when we converted to TypeScript, which was a little loose to start. A stricter pass on the TS would be awesome to work towards in v7.
